vixenNetwork.yml

```diff
@@ -0,0 +1,25 @@
+---
+name: "Vixen/Blacked/Tushy"
+sceneByURL:
+  - action: scrapeXPath
+    url:
+      - "https://www.blackedraw.com/"
+      - "https://www.blacked.com/"
+      - "https://www.tushy.com/"
+      - "https://www.tushyraw.com/"
+      - "https://www.vixen.com/"
+    scraper: sceneScraper
+xPathScrapers:
+  sceneScraper:
+    common:
+      $performer: //div[@data-test-component="VideoModels"]/a
+    scene:
+      Title: //h1[@data-test-component="VideoTitle"]
+      Performers:
+        Name: $performer
+        URL: $performer/@href
+      Details: //p[@data-test-component="VideoDescription"]
+      Date:
+        selector: //span[@data-test-component="ReleaseDateFormatted"]
+        parseDate: "January 2, 2006"
```

Comment:
```suggestion
      Date:
        selector: //span[@data-test-component="ReleaseDateFormatted"]
        parseDate: "January 2, 2006"
      Image: (//div[@data-test-component="ProgressiveImage"]/img/@src)[1]
```

This should scrape images as well

Reply:
Image links are generated through javascript, the above xpath doesn't work.

Reply:
Oh wow, the site doesn't load at all without javascript 
![TUSHY : Exclusive Anal HD Porn Videos 2020-05-13 10-23-23](https://user-images.githubusercontent.com/1358708/81824769-d9382f80-9503-11ea-88e2-5b7606202c39.jpg)

Is there a scrapping method that would work for images that are reliant on javascript?

Reply:
Have a look at the other scrapers.
First you identify the script section you need from the page source ( something like this `//script[contains(text(),'INITIAL_STATE')]` in this case) 
and then you regex replace from inside the url

Reply:
@bnkai When you get a chance can you try this one:
```
Image:
        selector: //script[contains(text(),'INITIAL_STATE')]
        replace:
          - regex: (?:{"name":"mainLandscape_1920x1080","width":1920,"height":1080,"src":")([^"]+)(?:.*)
          - with: $1
          - regex: \\u002F
          - with: /
```

Reply:
I've hit a wall in this one, I'll leave my progress here and maybe someone else can figure out the image field:

```
Image:
   selector: //script[contains(text(),'INITIAL_STATE')]/text()
   replace:
      - regex: .+(?:{"name":"mainLandscape_1920x1080","width":1920,"height":1080,"src":")(\w{5}:)(?:\\u002F\\u002F)([^\\u002F][\w-.]*)(?:\\u002F)([^\\u002F][\w]*)(?:\\u002F)([^\\u002F][\w]*)(?:\\u002F)([^\\u002F][\w]*)(?:\\u002F)([^\\u002F][\w]*)(?:\\u002F)([^\\u002F][\w-_.?=&]*).+
      with: $1//$2/$3/$4/$5/$6/$7
```

The `.+` on either end ensure the first image is always matched. In my initial testing these worked as the scenes I was testing on happened to have the main image listed first in their scripts. However on further testing I found it to be that images can appear in random quantity and order. If you were to remove the `.+` on either end you will see there are typically 1-7 matches for a scene with my regex. The main image and then any number of the suggested scenes on the page.

This is how the regex breaks down the matches:
![unknown](https://user-images.githubusercontent.com/1358708/82128358-933fcd80-9788-11ea-8ed8-06266fa53703.png)

I think the matches could be identified by the content in groups 4 or 7 as they seem to be more identifying, however I can't find data on the page that would confirm which ones of these are associated with the main scene.